plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/stock-status.block_theme.spec.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+import { test, expect } from '@woocommerce/e2e-playwright-utils';
+
+test.describe( 'Product Filter: Stock Status Block', async () => {
+	test.describe( 'frontend', () => {
+		test( 'Renders a checkbox list with the available stock statuses', async ( {
+			page,
+		} ) => {
+			await page.goto( '/product-filters-stock-status-block/' );
+
+			const stockStatuses = page.locator(
+				'.wc-block-components-checkbox__label'
+			);
+
+			await expect( stockStatuses ).toHaveCount( 2 );
+
+			await expect( stockStatuses.nth( 0 ) ).toHaveText( 'In stock' );
+			await expect( stockStatuses.nth( 1 ) ).toHaveText( 'Out of stock' );
+		} );
+
+		test( 'Selecting a stock status filters the list of products', async ( {
+			page,
+		} ) => {
+			await page.goto( '/product-filters-stock-status-block/' );
+
+			const stockStatusCheckboxes = page.locator(
+				'.wc-block-components-checkbox__input'
+			);
+
+			// Out of stock
+			stockStatusCheckboxes.nth( 1 ).check();
```

Comment:
What do you think about using `getByLabel()` here? It makes the code easier to read and it should be more resilient as we don't depend on a specific CSS selector.

```suggestion
			const outOfStockCheckbox = page.getByLabel( 'Out of stock' );
			outOfStockCheckbox.click();
```

Reply:
Great idea thanks will fix :)

Reply:
I've decided to migrate these tests to https://github.com/woocommerce/woocommerce/pull/44223 which is dynamically creating post content. I've incorporated these suggest changes in that PR. 🙇